src/apps/frontend/services/task.service.ts

```diff
@@ -0,0 +1,73 @@
+import { AccessToken, ApiError, ApiResponse } from '../types';
+import { JsonObject } from '../types/common-types';
+import { Task } from '../types/task';
+
+import APIService from './api.service';
+
+export default class TaskService extends APIService {
+  addTask = async (
+    title: string,
+    description: string,
+  ): Promise<ApiResponse<Task>> => {
+    try {
+      const userAccessToken = JSON.parse(
+        localStorage.getItem('access-token'),
+      ) as AccessToken;
+      const response = await this.apiClient.post(
+        '/tasks', { title, description },
+        {
+          headers: {
+            Authorization: `Bearer ${userAccessToken.token}`,
+          },
+        },
+      );
+      return new ApiResponse(new Task(response.data as JsonObject), undefined);
+    } catch (e) {
+      return new ApiResponse(undefined, new ApiError(e.response.data as JsonObject));
+    }
+  };
+
+  getTasks = async (): Promise<ApiResponse<Task[]>> => {
+    const userAccessToken = JSON.parse(
+      localStorage.getItem('access-token'),
+    ) as AccessToken;
+    return this.apiClient.get('/tasks', {
```

Comment:
Where are you returning type `ApiResponse`? There is no Javascript object being created, rather you are simply return JSON returned from server.

For example, if there is a custom function on `ApiResponse` defined, then the caller of the function will not be able to execute it on the return value type